Automattic/jetpack
# Boost: Add check for css gen library before generation attempt

Fixes #37091

## Proposed changes:

* Add a check before trying to use library;
* Add track event if there's something wrong with the library;
* Show a friendly error if something's wrong with the library:

> Critical CSS Generator library is either not found or invalid.

<details><summary>Before</summary>

![CleanShot 2024-05-08 at 16 28 14](https://github.com/Automattic/jetpack/assets/11799079/cf24e676-a1f9-45f2-9ae2-f08268d114fd)

</details>

<details><summary>Now</summary

![CleanShot 2024-05-08 at 16 26 12](https://github.com/Automattic/jetpack/assets/11799079/4c876dce-53dc-4074-a7d9-6e9dcf5917c2)

</details>

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Setup free Boost;
* Dequeue the critical css gen library (comment out this line - https://github.com/Automattic/jetpack/pull/37118/files#diff-a880940bee6572ef92319fcf362097099c74ab3ea6e03ba99fd6957407dd8abdR108);
* Enable critical css and try to generate. You should see the updated error.
